Source/SwiftLintFramework/Rules/Idiomatic/ConvenienceTypeRule.swift

```diff
@@ -60,12 +100,16 @@ public struct ConvenienceTypeRule: ASTRule, OptInRule, ConfigurationProviderRule
     public func validate(file: SwiftLintFile, kind: SwiftDeclarationKind,
                          dictionary: SourceKittenDictionary) -> [StyleViolation] {
         guard let offset = dictionary.offset,
+            // MARK: Is class or struct?
             [.class, .struct].contains(kind),
+            // MARK: Inherits from anything?
             dictionary.inheritedTypes.isEmpty,
+            // MARK: Substructure has members? i.e. is the type completely empty?
             !dictionary.substructure.isEmpty else {
```

Comment:
I'm not sure that // MARK: is needed here. 
Coment is enough

Also the comments itselves are too much.